dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -80,9 +80,10 @@ RtpsUdpDataLink::RtpsUdpDataLink(RtpsUdpTransport& transport,
   , job_queue_(DCPS::make_rch<DCPS::JobQueue>(reactor_task->get_reactor()))
   , multi_buff_(this, config.nak_depth_)
   , best_effort_heartbeat_count_(0)
-  , nack_reply_(this, &RtpsUdpDataLink::send_nack_replies,
-                config.nak_response_delay_)
-  , heartbeat_(reactor_task->interceptor(), config.heartbeat_period_, *this, &RtpsUdpDataLink::send_heartbeats)
+  , nack_reply_(reactor_task->interceptor(), *this, &RtpsUdpDataLink::send_nack_replies)
+  , expected_acks_(0)
```

Comment:
Does it make sense to move the timer and expected_acks_ field to the individual writers? I think we're (definitely) going to want it there eventually... without it, a writer publishing high speed data attached to a single reader will have his HB frequency decided by the acks to a writer publishing low speed status updates, etc.

Reply:
It absolutely does but it will require changing static discovery so I think we should take that up later.